### PR TITLE
Fixes Type.GetInterface fails to find implemented interface when `ignoreCase: true` is used

### LINKED ISSUE #6579: Type.GetInterface fails to find implemented interface when `ignoreCase: true` is used
## Steps to Reproduce

1. Run code below

```
using System;
using System.Collections.Generic;

public class Foo
{
    public static int Main(string[] args)
    {
        Console.WriteLine((typeof(Dictionary<string, object>).GetInterface("System.Collections.IDictionary", true) != null));
        return 0;
    }
}
```

## Current Behavior

Output: False

## Expected Behavior

// .NET returns True
Output: True

### On which platforms did you notice this

[ ] macOS
[ ] Linux
[ x] Windows

**Version Used**:

Recent mono master (perhaps a few weeks old)

### Stacktrace

```
Please paste the stacktrace here if available.
```

### PR DESCRIPTION
**Pull request by KonH**
Fixes #6579

### CODE REVIEW

File: mcs/class/corlib/Test/System/GetInterfaceTest.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+				type.GetInterface("System.Collections.IDictionary", false),
+				"strict named interface must be found (ignoreCase = false)"
+			);
+			Assert.NotNull(
+				type.GetInterface("System.Collections.IDictionary", true),
+				"strict named interface must be found (ignoreCase = true)"
+			);
+			Assert.NotNull(
+				type.GetInterface("System.Collections.Idictionary", true),
+				"interface, named in mixed case, must not be found (ignoreCase = false)"
```
Comment (lewurm) [inline mcs/class/corlib/Test/System/GetInterfaceTest.cs]: comment doesn't match test

File: mcs/class/referencesource/mscorlib/system/rttype.cs [hunk 2]
```diff
@@ -3414,8 +3414,14 @@ public override Type GetInterface(String fullname, bool ignoreCase)
 #if MONO
             List<RuntimeType> list = null;
             foreach (RuntimeType t in GetInterfaces ()) {
-                if (t.Name != name)
-                    continue;
+
+                if (ignoreCase) {
+                    if (t.Name.ToLowerInvariant() != name)
```
Comment (lewurm) [inline mcs/class/referencesource/mscorlib/system/rttype.cs]: shouldn't it be `t.Name.ToLowerInvariant () != name.ToLowerInvariant`?

Comment (marek-safar) [reply mcs/class/referencesource/mscorlib/system/rttype.cs]: Better to use special string comparer, see https://msdn.microsoft.com/en-us/library/e6883c06(v=vs.110).aspx

File: mcs/class/corlib/Test/System/GetInterfaceTest.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+// (C) Konstantin Khitrykh
+// 
+
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+
+namespace MonoTests.System {
+	[TestFixture]
+	public class GetInterfaceTest {
```
Comment (marek-safar) [inline mcs/class/corlib/Test/System/GetInterfaceTest.cs]: Not sure it's worth to add a new test file, just add the method to existing TypeTest.cs

File: mcs/class/corlib/Test/System/GetInterfaceTest.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+		[Test]
+		public void GetInterfaceTests() {
```
Comment (lambdageek) [inline mcs/class/corlib/Test/System/GetInterfaceTest.cs]: 1. This probably should go in `mcs/class/corlib/Test/System/TypeTest.cs` we try to group tests by the class of the method that they're testing.
2. The name of the test should be a little more descriptive: `GetInterfaceCaseInsensitiveTest`

File: mcs/class/referencesource/mscorlib/system/rttype.cs (hunk 2 above)
Comment (lambdageek) [reply mcs/class/referencesource/mscorlib/system/rttype.cs]: @lewurm RuntimeType.FilterHelper changes `name` to lower case.

Comment (akoeplinger) [reply mcs/class/referencesource/mscorlib/system/rttype.cs]: I think we could simplify this to:

```csharp
if (!t.Name.Equals(name, ignoreCase ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal))
  continue;
```

**Comment by akoeplinger** [comment]
@monojenkins build

**Comment by KonH** [comment]
https://github.com/mono/mono/pull/7464/commits/1b12f154636facc14b2bc7c0ca104971a78d6937 - here I have implemented proposed changes, do I need additional actions in pull request?

**Comment by lambdageek** [comment]
Nope, that's all.  Thanks again, @KonH.   We'll watch the build results and merge it when they're ready.

**Pull request merged**